### PR TITLE
Fixing SnapshotResiliencyTests.testTransportGetSnapshotsAction

### LINKED ISSUE #9186: [Flaky Test]SnapshotResiliencyTests.testTransportGetSnapshotsAction
**Describe the bug**
https://build.ci.opensearch.org/job/gradle-check/22110/
Tests with failures:
 - org.opensearch.snapshots.SnapshotResiliencyTests.testTransportGetSnapshotsAction

```

java.lang.AssertionError: Blobstore is closed already
	at __randomizedtesting.SeedInfo.seed([A26299704FDB6338:1BB7497D88CBC8D2]:0)
	at org.opensearch.snapshots.mockstore.MockEventuallyConsistentRepository$MockBlobStore.ensureNotClosed(MockEventuallyConsistentRepository.java:193)
	at org.opensearch.snapshots.mockstore.MockEventuallyConsistentRepository$MockBlobStore$MockBlobContainer.delete(MockEventuallyConsistentRepository.java:274)
	at org.opensearch.repositories.blobstore.BlobStoreRepository.endVerification(BlobStoreRepository.java:1839)
	at org.opensearch.repositories.RepositoriesService$3.lambda$doRun$1(RepositoriesService.java:375)
	at org.opensearch.cluster.coordination.AbstractCoordinatorTestCase$1.run(AbstractCoordinatorTestCase.java:1468)
	at org.opensearch.cluster.coordination.DeterministicTaskQueue.runTask(DeterministicTaskQueue.java:146)
	at org.opensearch.cluster.coordination.DeterministicTaskQueue.runRandomTask(DeterministicTaskQueue.java:140)
	at org.opensearch.cluster.coordination.DeterministicTaskQueue.runAllRunnableTasks(DeterministicTaskQueue.java:90)
	at org.opensearch.snapshots.SnapshotResiliencyTests.testTransportGetSnapshotsAction(SnapshotResiliencyTests.java:822)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
```

**To Reproduce**
```
REPRODUCE WITH: ./gradlew ':server:test' --tests "org.opensearch.snapshots.SnapshotResiliencyTests.testTransportGetSnapshotsAction" -Dtests.seed=A26299704FDB6338 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=en-MT -Dtests.timezone=Pacific/Pago_Pago -Druntime.java=20
```

**Expected behavior**

**Plugins**
Please list all plugins currently enabled.

**Screenshots**

**Host/Environment (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by indrajohn7**
### Description
This PR resolves the flaky test: `testTransportGetSnapshotsAction` with a single createSnapshotResponseStepListener object.  

### Related Issues  
Resolves #9186

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -51,6 +51,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Replace the deprecated IndexReader APIs with new storedFields() & termVectors() ([#7792](https://github.com/opensearch-project/OpenSearch/pull/7792))
 - [Remote Store] Restrict user override for remote store index level settings ([#8812](https://github.com/opensearch-project/OpenSearch/pull/8812))
 - Removed blocking wait in TransportGetSnapshotsAction which was exhausting generic threadpool ([#8377](https://github.com/opensearch-project/OpenSearch/pull/8377))
+- Fixed Flaky test testTransportGetSnapshotsAction ([#9189](https://github.com/opensearch-project/OpenSearch/pull/9189))
```
Comment (shwetathareja) [inline CHANGELOG.md:54]: we dont need change log.

File: CHANGELOG.md [hunk 2]
```diff
@@ -129,4 +130,4 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 ### Security
 
 [Unreleased 3.0]: https://github.com/opensearch-project/OpenSearch/compare/2.x...HEAD
-[Unreleased 2.x]: https://github.com/opensearch-project/OpenSearch/compare/2.10...2.x
\ No newline at end of file
+[Unreleased 2.x]: https://github.com/opensearch-project/OpenSearch/compare/2.10...2.x
```
Comment (shwetathareja) [inline CHANGELOG.md:133]: unrelated?

File: server/src/test/java/org/opensearch/snapshots/SnapshotResiliencyTests.java [hunk 3]
```diff
@@ -798,43 +798,41 @@ public void testTransportGetSnapshotsAction() {
...
+                assertNotNull("Snapshot list should not be null", repoSnapshotResponse.getSnapshots());
+                assertThat(repoSnapshotResponse.getSnapshots(), hasSize(1));
+                List<SnapshotInfo> snapshotInfos = repoSnapshotResponse.getSnapshots();
+                SnapshotInfo snapshotInfo = snapshotInfos.get(0);
                 assertEquals(SnapshotState.SUCCESS, snapshotInfo.state());
                 assertEquals(0, snapshotInfo.failedShards());
-                assertTrue(Arrays.stream(snapshotsList).anyMatch(snapshotInfo.snapshotId().getName()::equals));
-            }
-        }, exception -> { throw new AssertionError(exception); }));
+                assertTrue(snapshotInfo.snapshotId().getName().equals(snapshotsList[0]));
```
Comment (shwetathareja) [inline server/src/test/java/org/opensearch/snapshots/SnapshotResiliencyTests.java:833]: assertEquals

File: server/src/test/java/org/opensearch/snapshots/SnapshotResiliencyTests.java [hunk 4]
```diff
@@ -798,43 +798,41 @@ public void testTransportGetSnapshotsAction() {
         setupTestCluster(randomFrom(1, 3, 5), randomIntBetween(2, 10));
 
         String repoName = "repo";
-        final String[] snapshotsList = { "snapshot-1", "snapshot-2" };
-        final String[] indexList = { "index-1", "index-2" };
+        final String[] snapshotsList = { "snapshot-1" };
+        final String index = "index-1";
         final int shards = randomIntBetween(1, 10);
```
[the remark this answers is not in the document]
Comment (indrajohn7) [reply server/src/test/java/org/opensearch/snapshots/SnapshotResiliencyTests.java:803]: We already have tests with multiple snapshots and indices. Here we wanted to confirm that the `transportGetSnapshotsAction` which is listener based is properly invoked. The original fix is in making the single `createSnapshotResponseStepListener` object and also in stead of `deterministicTaskQueue`, now we will be using the `continueOrDie` workflow which will be based on the snapshot response listener object.

[the remark this answers is not in the document]
Comment (Bukhtawar) [reply server/src/test/java/org/opensearch/snapshots/SnapshotResiliencyTests.java:803]: The test name suggests `transportGetSnapshot"s"Action` and hence the question

[the remark this answers is not in the document]
Comment (shwetathareja) [reply server/src/test/java/org/opensearch/snapshots/SnapshotResiliencyTests.java:803]: There was no need for ```deterministicTaskQueue.runAllRunnableTasks();``` as there are no cluster state change listeners added. Instead, it should simply wait for ```createSnapshotResponseStepListener``` to be triggered and run the transport call and assertions. Moving to single snapshot only helped where it need not deal with multiple listeners and keep test simple. But still need to see what was the race condition where ```deterministicTaskQueue.runAllRunnableTasks()``` this was called before listener.

**Review by shwetathareja — APPROVED** [review]
